build/scripts/build-utils.ps1

```diff
@@ -6,10 +6,40 @@ $ErrorActionPreference="Stop"
 # Declare a number of useful variables for other scripts to use
 [string]$repoDir = Resolve-Path (Join-Path $PSScriptRoot "..\..")
 [string]$binariesDir = Join-Path $repoDir "Binaries"
-[string]$scriptDir = $PSScriptRoot
+
+# Handy function for executing a command in powershell and throwing if it 
+# fails.  
+# 
+# Original sample came from: http://jameskovacs.com/2010/02/25/the-exec-problem/
+function Exec([scriptblock]$cmd, [string]$errorMessage = "Error executing command: " + $cmd) { 
```

Comment:
For external commands, why aren't we checking `$LastExitCode` and handling that appropriately on a per executable basis?